src/common/Gateway.sol

```diff
@@ -286,16 +271,19 @@ contract Gateway is Auth, IGateway, Recoverable {
         emit PrepareMessage(centrifugeId, poolId, message);
 
         if (isBatching) {
-            bytes storage previousMessage = outboundBatch[centrifugeId][poolId];
+            bytes32 batchSlot = keccak256(abi.encode("outboundBatch", centrifugeId, poolId));
+            bytes memory previousMessage = TransientBytesLib.get(batchSlot);
 
-            batchGasLimit[centrifugeId][poolId] += gasService.gasLimit(centrifugeId, message);
-            require(batchGasLimit[centrifugeId][poolId] <= gasService.maxBatchSize(centrifugeId), ExceedsMaxBatchSize());
+            bytes32 gasLimitSlot = keccak256(abi.encode("batchGasLimit", centrifugeId, poolId));
+            uint128 newGasLimit = gasLimitSlot.tloadUint128() + gasService.gasLimit(centrifugeId, message);
+            require(newGasLimit <= gasService.maxBatchSize(centrifugeId), ExceedsMaxBatchSize());
+            gasLimitSlot.tstore(uint256(newGasLimit));
 
             if (previousMessage.length == 0) {
-                batchLocators.push(BatchLocator(centrifugeId, poolId));
-                outboundBatch[centrifugeId][poolId] = message;
+                TransientArrayLib.push(BATCH_LOCATORS_SLOT, bytes32(abi.encodePacked(bytes2(centrifugeId), bytes8(poolId.raw()))));
+                TransientBytesLib.set(batchSlot, message);
             } else {
-                outboundBatch[centrifugeId][poolId] = bytes.concat(previousMessage, message);
+                TransientBytesLib.set(batchSlot, bytes.concat(previousMessage, message));
```

Comment:
I think it should not be complicated to do a `TransientBytesLib.append()` to avoid concatenating the message here, and just insert the new message at the end of the bytes.

Will try to do it if I have time before Wednesday!

Reply:
Quickly looked at it, it is a little complex since you need to either add just new chunks, or edit the last chunk if `length % 32 != 0`.

Reply:
Yeah, I think you should slice() the first chunk and the last stored chunk, join both, and then continue as usual